griptape/tools/google_docs/tool.py

```diff
@@ -0,0 +1,211 @@
+from __future__ import annotations
+import logging
+from attr import field, define
+from schema import Schema, Optional
+from griptape.artifacts import ErrorArtifact, InfoArtifact, BlobArtifact, ListArtifact
+from griptape.utils.decorators import activity
+from griptape.tools import BaseGoogleClient
+from googleapiclient.discovery import Resource
+
+@define
+class GoogleDocsClient(BaseGoogleClient):
+    DOCS_SCOPES = ["https://www.googleapis.com/auth/documents"]
+
+    DRIVE_SCOPES = ["https://www.googleapis.com/auth/drive.file"]
+
+    owner_email: str = field(default=None)
+
+    @activity(
+        config={
+            "description": "Can be used to create a new Google Doc in Drive",
+            "schema": Schema({"file_name": str}),
+        }
+    )
+    def create_google_doc(self, file_name: str) -> InfoArtifact | ErrorArtifact:
+        try:
+            service = self._build_client(self.DOCS_SCOPES, "v1", "docs")
+            doc = service.documents().create(body={"title": file_name}).execute()
+            return InfoArtifact(f"Google Doc '{file_name}' created with ID: {doc['documentId']}")
+
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"Error creating Google Doc: {e}")
+
+    @activity(
+        config={
+            "description": "Can be used to append text to a Google Doc",
+            "schema": Schema({"file_path": str, "text": str}),
+        }
+    )
+    def append_text_to_google_doc(self, file_path: str, text: str) -> InfoArtifact | ErrorArtifact:
+        try:
+            service = self._build_client(self.DOCS_SCOPES, "v1", "docs")
+            drive_service = self._build_client(self.DRIVE_SCOPES, "v3", "drive")
+            document_id = self._path_to_file_id(drive_service, file_path)
+            if document_id:
+                doc = service.documents().get(documentId=document_id).execute()
+                content = doc["body"]["content"]
+                last_text = content[-1]["paragraph"]["elements"][-1]["textRun"]["content"]
+                append_index = content[-1]["endIndex"]
+                if last_text.endswith("\n"):
+                    append_index -= 1
+
+                requests = [{"insertText": {"location": {"index": append_index}, "text": text}}]
+
+                doc = service.documents().batchUpdate(documentId=document_id, body={"requests": requests}).execute()
+                return InfoArtifact("Text appended successfully")
+            else:
+                return ErrorArtifact(f"Error: File not found for path {file_path}")
```

Comment:
Update error string to `error appending to Google Doc, file not found for path {file_path}`

Reply:
Done!